rllib/tests/test_algorithm_save_load_checkpoint_learner.py

```diff
@@ -0,0 +1,90 @@
+import tempfile
+import unittest
+
+import ray
+from ray.rllib.utils.test_utils import check, framework_iterator
+from ray.rllib.algorithms.ppo import PPOConfig
+from ray.rllib.policy.sample_batch import DEFAULT_POLICY_ID
+from ray.rllib.utils.metrics.learner_info import LEARNER_INFO, LEARNER_STATS_KEY
+
+
+algorithms_and_configs = {
+    "PPO": (PPOConfig().training(train_batch_size=2, sgd_minibatch_size=2))
+}
+
+
+@ray.remote
+def save_and_train(algo_cfg, env, tmpdir):
+    algo_cfg = (
+        algo_cfg.training(_enable_learner_api=True)
+        .rl_module(_enable_rl_module_api=True)
+        .rollouts(num_rollout_workers=0)
+        .reporting(min_time_s_per_iteration=0, min_sample_timesteps_per_iteration=1)
+        .debugging(seed=10)
+    )
+    algo1 = algo_cfg.environment(env).build()
+
+    print(tmpdir)
+    tmpdir = str(tmpdir)
+    algo1.save_checkpoint(tmpdir)
+    print("======= training algo 1 =======")
+    algo1.train()
+    results_algo1 = algo1.train()
+    return results_algo1["info"][LEARNER_INFO][DEFAULT_POLICY_ID][LEARNER_STATS_KEY]
+
+
+@ray.remote
+def load_and_train(algo_cfg, env, tmpdir):
+    algo_cfg = (
+        algo_cfg.training(_enable_learner_api=True)
+        .rl_module(_enable_rl_module_api=True)
+        .rollouts(num_rollout_workers=0)
+        .reporting(min_time_s_per_iteration=0, min_sample_timesteps_per_iteration=1)
```

Comment:
setting min time spend per each iteration results into stochasticity of the outcome, is this intentional?

Reply:
I intentionally did this because I wanted results to only come after the min_sample_timesteps_per_iteration was met. 
Is this not whats happening here? 

does it need to be like 1 at minimum or something like that so that I can get the outcome where results reporting is based only on sample timesteps?

Reply:
ah it turns out this needs to stay 0. the reason is that I want my results reporting to be purely deicated by the number of timesteps that is sampled. If I set this to say like 1, then if this criteria is met before the sample time steps criteria then results will be reported. This will result in stochastic result reporting.